server/datastore/mysql/apple_mdm_test.go

```diff
@@ -6558,170 +6558,381 @@ func testAppleMDMVPPTokensCRUD(t *testing.T, ds *Datastore) {
+	assert.NoError(t, err)
+	assert.Equal(t, tokTeam2.ID, tokTeam2.ID)
+	assert.Len(t, tokTeam2.Teams, 2)
+	assert.Contains(t, tokTeam2.Teams, fleet.TeamTuple{ID: team2.ID, Name: team2.Name})
+	assert.Contains(t, tokTeam2.Teams, fleet.TeamTuple{ID: 0, Name: fleet.TeamNameNoTeam})
 
-	err = ds.DeleteVPPToken(ctx, toknone.ID)
-	require.NoError(t, err)
+	////
+	err = ds.DeleteVPPToken(ctx, tokTeam.ID)
+	assert.NoError(t, err)
 
-	bynone, err := ds.GetVPPTokenByTeamID(ctx, nil)
-	require.NoError(t, err)
-	assert.Equal(t, tokall.ID, bynone.ID)
-	assert.Equal(t, tokall.TeamID, bynone.TeamID)
-	assert.Equal(t, tokall.NullTeam, bynone.NullTeam)
-	assert.Equal(t, tokall.Token, bynone.Token)
-	assert.Equal(t, tokall.Location, bynone.Location)
-	assert.Equal(t, tokall.OrgName, bynone.OrgName)
-	assert.Equal(t, tokall.RenewDate, bynone.RenewDate)
+	tokNil, err = ds.GetVPPTokenByTeamID(ctx, nil)
+	assert.NoError(t, err)
+	assert.Equal(t, tokTeams.ID, tokNil.ID)
 
-	// Removing the all teams token means error
-	err = ds.DeleteVPPToken(ctx, tokall.ID)
-	require.NoError(t, err)
+	tokTeam1, err = ds.GetVPPTokenByTeamID(ctx, &team.ID)
+	assert.NoError(t, err)
+	assert.Equal(t, tokAll.ID, tokTeam1.ID)
 
-	_, err = ds.GetVPPTokenByTeamID(ctx, nil)
-	require.Error(t, err)
-	require.ErrorIs(t, err, sql.ErrNoRows)
+	tokTeam2, err = ds.GetVPPTokenByTeamID(ctx, &team2.ID)
+	assert.NoError(t, err)
+	assert.Equal(t, tokTeams.ID, tokTeam2.ID)
 
-	_, err = ds.GetVPPTokenByTeamID(ctx, tok2.TeamID)
-	require.Error(t, err)
-	require.ErrorIs(t, err, sql.ErrNoRows)
+	////
+	err = ds.DeleteVPPToken(ctx, tokTeams.ID)
+	assert.NoError(t, err)
+
+	tokNil, err = ds.GetVPPTokenByTeamID(ctx, nil)
+	assert.NoError(t, err)
+	assert.Equal(t, tokAll.ID, tokNil.ID)
+
+	tokTeam1, err = ds.GetVPPTokenByTeamID(ctx, &team.ID)
+	assert.NoError(t, err)
+	assert.Equal(t, tokAll.ID, tokTeam1.ID)
+
+	tokTeam2, err = ds.GetVPPTokenByTeamID(ctx, &team2.ID)
+	assert.NoError(t, err)
+	assert.Equal(t, tokAll.ID, tokTeam2.ID)
+
+	////
+	err = ds.DeleteVPPToken(ctx, tokAll.ID)
+	assert.NoError(t, err)
+
+	tokNil, err = ds.GetVPPTokenByTeamID(ctx, nil)
+	assert.Error(t, err)
+	assert.True(t, fleet.IsNotFound(err))
+
+	tokTeam1, err = ds.GetVPPTokenByTeamID(ctx, &team.ID)
+	assert.Error(t, err)
+	assert.True(t, fleet.IsNotFound(err))
+
+	tokTeam2, err = ds.GetVPPTokenByTeamID(ctx, &team2.ID)
+	assert.Error(t, err)
+	assert.True(t, fleet.IsNotFound(err))
+
+	////
+	_, err = ds.UpdateVPPTokenTeams(ctx, tokNone.ID, []uint{0, team.ID, team2.ID})
+	assert.NoError(t, err)
+
+	tokNil, err = ds.GetVPPTokenByTeamID(ctx, nil)
+	assert.NoError(t, err)
+	assert.Equal(t, tokNone.ID, tokNil.ID)
+
+	tokTeam1, err = ds.GetVPPTokenByTeamID(ctx, &team.ID)
+	assert.NoError(t, err)
+	assert.Equal(t, tokNone.ID, tokTeam1.ID)
+
+	tokTeam2, err = ds.GetVPPTokenByTeamID(ctx, &team2.ID)
+	assert.NoError(t, err)
+	assert.Equal(t, tokNone.ID, tokTeam2.ID)
+
+	////
+	err = ds.DeleteVPPToken(ctx, tokNone.ID)
+	assert.NoError(t, err)
```

Comment:
Nit, but for clarity this should probably be moved to `vpp_test.go` to be alongside the methods implementation in `vpp.go`.